src/js/common/validator.js

```diff
@@ -148,21 +149,75 @@ var fluid  = fluid  || {};
 
     /**
      *
-     * Validate material against a "GPII Schema System" schema.
+     * Validate material against a "GPII Schema System" schema using a precompiled AJV "validator".
      *
-     * @param {Any} toValidate - The material to be validated.
+     * @param {Object} that - The validator component.
      * @param {GssSchema} gssSchema - A GSS schema definition.
-     * @param {Object} ajvOptions - Optional arguments to pass to the underlying AJV validator.
+     * @param {Any} toValidate - The material to be validated.
+     * @param {String} [schemaHash] - An optional precomputed hash of the schema.
      * @return {gssValidationError} - An object that describes the results of validation.  The `isValid` property will be `true` if the data is valid, or `false` otherwise.  The `isError` property will be set to `true` if there are validation errors.
      *
      */
-    gpii.schema.validator.validate = function (toValidate, gssSchema, ajvOptions) {
+    gpii.schema.validator.validate = function (that, gssSchema, toValidate, schemaHash) {
+        schemaHash = schemaHash || gpii.schema.stringify(gssSchema);
+
+        var validator = that.validatorsByHash[schemaHash];
+        if (!validator) {
+            try {
+                validator = that.cacheSchema(gssSchema, schemaHash);
+            }
+            catch (compileErrors) {
+                return { isError: true, message: "Error compiling GSS Schema.", errors: compileErrors};
+            }
+        }
+
+        validator(toValidate);
+        return validator.standardiseAjvErrors();
+    };
+
+    /**
+     *
+     * Add a single GSS schema to the cache.
+     *
+     * @param {Object} that - The validator component itself.
+     * @param {GssSchema} gssSchema - The original GSS schema.
+     * @param {String} [schemaHash] - An optional precomputed hash of the schema.
+     * @return {Object} - The compiled validator created from the GSS schema.
+     *
+     */
+    gpii.schema.validator.cacheSchema = function (that, gssSchema, schemaHash) {
```

Comment:
Nice addition - if, as you seem to be allowing in all of these functions, that the user may have computed the hash already, best to produce a generic function name (e.g. gpii.schema.hashSchema, which will currently just be an alias for gpii.schema.stringify) which they might have used to do so -

Reply:
Good idea, I added this in the validator file so that it could take advantage of the custom JSDoc object type.  I also updated the docs.